input/script_utils.go

```diff
@@ -141,6 +142,40 @@ func GenFundingPkScript(aPub, bPub []byte, amt int64) ([]byte, *wire.TxOut, erro
 	return witnessScript, wire.NewTxOut(amt, pkScript), nil
 }
 
+// GenTaprootFundingScript constructs the taproot-native funding output that
+// uses musig2 to create a single aggregated key to anchor the channel.
+func GenTaprootFundingScript(aPub, bPub *btcec.PublicKey,
+	amt int64) ([]byte, *wire.TxOut, error) {
+
+	// Similar to the existing p2wsh funding script, we'll always make sure
+	// we sort the keys before any major operations. In order to ensure
+	// that there's no other way this output can be spent, we'll use a BIP
+	// 86 tweak here during aggregation.
+	//
+	// TODO(roasbeef): revisit if BIP 86 is needed here?
```

Comment:
So the BIP 86 usage here (another tweak passed into the musig2 key aggregation) strongly binds the generated key to the LN channel context. This is because a 3rd party can verify that it's just a 2-of-2 musig and doesn't commit to any script paths at all. This is matches the current validation today for segwit v0 channels, as during validation we check that the script is only that 2-of-2 multi-sig and nothing more. 

Some devs are into the "gossip v2" concept wherein we'd relax this validation, as people could use any UTXO they can sign for to "advertise" a Lighting channel. In this context, then the output would be able to commit to an actual script path, as maybe it's some multi-sig scheme that lets others rent UTXOs or w/e. For this, we'd then remove this extra BIP 86 tweak as the _actual_ tweak would be the tapscript root. 

Another use case that wants an actual tapscript tweak here is Taro. In this context, we'd be committing to the taro asset tree in the leaf. 

This only really mainfests itself on the gossip layer though (advertised channels), as the verifiers need to know what to expect as the taptweak. I think we can make things general here though by explicitly specifying the taptweak in the Gossip 1.5 messages. Then verifies can either require that this is the BIP 86 tweak (binds the channel to just schnorr 2-of-2 multi-sig) or that it can be anything.

Reply:
Okay got it. So BIP 86 now doesn't rule out any future extensions and still allows gossip 1.5 verification. Sounds good. Perhaps useful to record the rationale in the bolt.